subsys/zigbee/lib/zigbee_app_utils/zigbee_app_utils.c

```diff
@@ -618,6 +654,52 @@ zb_ret_t zigbee_default_signal_handler(zb_bufid_t bufid)
 		/* Obsolete signals, used for pre-R21 ZBOSS API. Ignore. */
 		break;
 
+	case ZB_BDB_SIGNAL_TC_REJOIN_DONE:
+		/* This signal informs that TC rejoin is completed.
+		 * The signal status indicates if the device has successfully
+		 * rejoined the network.
+		 *
+		 * Next step: if the device implement Zigbee router or
+		 *            end device, and the TC rejoijn has failed,
+		 *            perform restart the generic rejoin procedure.
+		 */
+		if (IS_ENABLED(CONFIG_ZIGBEE_TC_REJOIN_ENABLED)) {
+			if (status == RET_OK) {
+				zb_ext_pan_id_t extended_pan_id;
+				char ieee_addr_buf[IEEE_ADDR_BUF_SIZE] = { 0 };
+				int addr_len;
+
+				zb_get_extended_pan_id(extended_pan_id);
+				addr_len = ieee_addr_to_str(ieee_addr_buf,
+								sizeof(ieee_addr_buf),
+								extended_pan_id);
+				if (addr_len < 0) {
+					strcpy(ieee_addr_buf, "unknown");
+				}
+
+				LOG_INF("Joined network successfully after TC rejoin"
+						" (Extended PAN ID: %s, PAN ID: 0x%04hx)",
+					log_strdup(ieee_addr_buf),
+					ZB_PIBCACHE_PAN_ID());
+				/* Device has joined the network so stop the network
+				 * rejoin procedure.
+				 */
+				if (role != ZB_NWK_DEVICE_TYPE_COORDINATOR) {
+					stop_network_rejoin(ZB_FALSE);
+				}
+			} else {
+				if (role != ZB_NWK_DEVICE_TYPE_COORDINATOR) {
+					LOG_INF("TC Rejoin was not successful (status: %d)",
+						status);
+					start_network_rejoin();
+				} else {
+					LOG_INF("TC Rejoin failed on Zigbee coordinator"
+							" (status: %d)", status);
+				}
+			}
+			break;
+		}
```

Comment:
Please add a comment, indicating that the fall-through to the default case is intentional.

Reply:
Added